### PR TITLE
Update: Add date-dimension filter builder for dimensions of datatype date

### PR DESCRIPTION
**Pull request by mattgraham1995**
Uses the navi-date-picker to allow selection of date/interval values for dimensions with date values. 

The supported operators are Since, Before, and Between.

screen shot 2018-11-01 at 6 40 30 pm

screen shot 2018-11-02 at 11 24 40 am

screen shot 2018-11-02 at 11 25 11 am

### CODE REVIEW

File: packages/core/addon/models/bard-request/fragments/filter.js [hunk 1]
```diff
@@ -5,12 +5,13 @@
...
 import { validator, buildValidations } from 'ember-cp-validations';
+import { computed, get, set } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { A as arr } from '@ember/array';
+import RSVP from 'rsvp';
 
-const { Fragment } = MF;
-
-const { computed, get, set } = Ember,
+const { Fragment } = MF,
```
Comment (balajimanoharan) [inline packages/core/addon/models/bard-request/fragments/filter.js]: `import { Fragment } from 'model-fragments';`

File: packages/core/addon/models/bard-request/fragments/filter.js [hunk 2]
```diff
@@ -37,10 +38,10 @@ const { computed, get, set } = Ember,
 export default Fragment.extend(Validations, {
   dimension: DS.attr('dimension'),
   operator: DS.attr('string', { defaultValue: 'in' }),
-  rawValues: DS.attr({ defaultValue: () => Ember.A([]) }),
+  rawValues: DS.attr({ defaultValue: () => arr([]) }),
   field: DS.attr('string', { defaultValue: 'id' }),
 
-  dimensionService: Ember.inject.service('bard-dimensions'),
+  dimensionService: service('bard-dimensions'),
```
Comment (balajimanoharan) [inline packages/core/addon/models/bard-request/fragments/filter.js]: can we jsdoc this?

File: packages/data/addon/mirage/fixtures/bard-meta-dimensions.js [hunk 3]
```diff
@@ -5,187 +5,221 @@ export default {
...
       fields: [
         { name: 'id', tags: ['id'] },
         { name: 'desc', tags: ['description'] },
         { name: 'key', tags: ['primaryKey'] }
       ]
+    },
+    {
+      name: 'userSignupDate',
+      longName: 'User Signup Date',
+      cardinality: 39896,
```
Comment (balajimanoharan) [inline packages/data/addon/mirage/fixtures/bard-meta-dimensions.js]: so precise.

File: packages/reports/addon/components/filter-collection.js [hunk 4]
```diff
@@ -35,8 +35,11 @@ export default Ember.Component.extend({
     });
 
     let dimFilters = get(this, 'request.filters').map(filter => {
+      let dimensionDataType = get(filter, 'dimension.datatype'),
```
Comment (balajimanoharan) [inline packages/reports/addon/components/filter-collection.js]: const

File: packages/reports/addon/components/filter-values/date.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import layout from '../../templates/components/filter-values/date';
+import { computed, get } from '@ember/object';
+
+export default Component.extend({
+  layout,
+
+  /**
+   * @property {Array} classNames
+   */
+  classNames: ['date'],
```
Comment (balajimanoharan) [inline packages/reports/addon/components/filter-values/date.js]: I feel like we need a better name than date

File: packages/reports/addon/components/filter-values/date.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+   */
+  selectedDate: null,
+
+  /**
+   * @property {String} displayedDate
+   */
+  displayedDate: computed('filter.values', function() {
+    let filterDate = get(this, 'filter.values.firstObject');
+
+    return filterDate ? Moment(filterDate).format('MMM DD, YYYY') : 'Select date';
```
Comment (balajimanoharan) [inline packages/reports/addon/components/filter-values/date.js]: Should `Select date` be a placeholder instead?

File: packages/reports/addon/components/filter-values/date.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  }),
+
+  actions: {
+    /**
+     * @action setDate
+     * @param {Date} date - new date to set in filter
+     */
+    setDate(date) {
+      this.attrs.onUpdateFilter({
+        values: [Moment(date).format('YYYY-MM-DD')]
```
Comment (balajimanoharan) [inline packages/reports/addon/components/filter-values/date.js]: Is the difference between the display format and this intentional?

File: packages/reports/addon/components/filter-values/date.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+      this.set('selectedDate', savedDate);
+    },
+
+    /**
+     * @action closeDropdown
+     * @param {Object} dropdown - the ember-basic-dropdown object
+     */
+    closeDropdown(dropdown) {
+      dropdown.actions.close();
```
Comment (balajimanoharan) [inline packages/reports/addon/components/filter-values/date.js]: Can we do this in the template instead?
like we have done here.
https://github.com/yahoo/navi/blob/01ff1618019b137506ee005d22851364aae2c236/packages/reports/addon/templates/components/metric-filter-config.hbs#L14

File: packages/reports/addon/components/filter-values/date.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+  /**
+   * @property {Array} classNames
+   */
+  classNames: ['date'],
+
+  /**
+   * @property {Moment} selectedDate
+   */
+  selectedDate: null,
```
Comment (balajimanoharan) [inline packages/reports/addon/components/filter-values/date.js]: Is this a local state? How is this different from `displayedDate`?

File: packages/data/addon/mirage/fixtures/bard-meta-dimensions.js (hunk 3 above)
Comment (mattgraham1995) [reply packages/data/addon/mirage/fixtures/bard-meta-dimensions.js]: 🎯

File: packages/reports/addon/components/filter-values/date.js (hunk 5 above)
Comment (mattgraham1995) [reply packages/reports/addon/components/filter-values/date.js]: `date-range` is the existing one so I figured `date` was a logical choice for a single date.

File: packages/reports/addon/components/filter-values/date.js (hunk 7 above)
Comment (mattgraham1995) [reply packages/reports/addon/components/filter-values/date.js]: Display format is literally only for display whereas this string is what's passed to the querybuilder and used in the filter param

File: packages/reports/addon/components/filter-values/date.js (hunk 9 above)
Comment (mattgraham1995) [reply packages/reports/addon/components/filter-values/date.js]: This is the local state used by navi-date-picker. displayedDate is a string and selectedDate is a moment object

**Comment by mattgraham1995** [comment]
Splitting into multiple PRs.

**Pull request closed without merging**